sensor/common/metrics/metrics.go

```diff
@@ -153,6 +153,13 @@ var (
 		Help:      "A counter for the total number of events sent from Sensor to Central",
 	}, []string{"Action", "ResourceType", "Type"})
 
+	sensorGRPCMaxMessageSize = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "max_message_size_sent",
+		Help:      "A gauge for maximum message size seen when sending messages in sensor",
+	}, []string{"Type", "EventType"})
+
```

Comment:
I have been thinking about this and I tend even more to making it a histogram.
Why?
- Computing the maximum is done by us in code using additional map - it is extra code to maintain,
- When sensor/central restarts, we will loose the current maxima and will report a new maximum to prometheus. This means that looking at a diag bundle after a restart will tell us nothing about the message sizes from before the restart. (One option to mitigate this would be to look at the pod-ID in the metrics).

Using HistogramVec instead would allow us to avoid those issues. The only trick is to plan the bucket sizes generously, so that we can capture various message sizes.

As a plan-minimum, I would like to confirm that we are able to get the metrics values from prometheus about the historical pods that were restarted. As this may not be optimal (too many metrics for past pods), I would suggest moving to the `HistogramVec` as this costs us nearly nothing.

Reply:
> When sensor/central restarts, we will loose the current maxima and will report a new maximum to prometheus.

That's a good point. I'd be inclined to use a histogram too.

> I would like to confirm that we are able to get the metrics values from prometheus about the historical pods that were restarted

I don't know for sure but I think prometheus removes that information. At least for some metrics, I haven't seen information of previous pods. It would be good to confirm this.

Reply:
> This means that looking at a diag bundle after a restart will tell us nothing about the message sizes from before the restart

I would assume this is the same in histogram?

I can change it to histogram and test.

Reply:
A test would be great - could you maybe add histogram along the max gauge (only for the test), so that we can directly compare?